packages/smooth_app/lib/pages/user_management/forgot_password_page.dart

```diff
@@ -0,0 +1,233 @@
+import 'dart:io' show Platform;
+
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:open_mail_app/open_mail_app.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:provider/provider.dart';
+import 'package:smooth_app/themes/theme_provider.dart';
+import 'package:smooth_ui_library/widgets/smooth_card.dart';
+import 'package:smooth_ui_library/widgets/smooth_text_form_field.dart';
+
+class ForgotPasswordPage extends StatefulWidget {
+  const ForgotPasswordPage({Key? key}) : super(key: key);
+
+  @override
+  State<ForgotPasswordPage> createState() => _ForgotPasswordPageState();
+}
+
+class _ForgotPasswordPageState extends State<ForgotPasswordPage> {
+  static Color textFieldBackgroundColor = const Color.fromARGB(
+    255,
+    240,
+    240,
+    240,
+  );
+
+  final GlobalKey<FormState> _formKey = GlobalKey<FormState>();
+  final TextEditingController userIdController = TextEditingController();
+
+  bool _send = false;
+  bool _runningQuery = false;
+  late String _message;
+
+  Future<void> _resetPassword() async {
+    final AppLocalizations appLocalizations = AppLocalizations.of(context)!;
+    if (!_formKey.currentState!.validate()) {
+      return;
+    }
+    setState(() => _runningQuery = true);
+
+    final Status status;
+    try {
+      status = await OpenFoodAPIClient.resetPassword(userIdController.text);
+    } catch (e) {
+      throw Exception(e);
+    }
+
+    if (status.status == 200) {
+      _send = true;
+      _message = appLocalizations.reset_password_done;
+    } else if (status.status == 400) {
+      _message = appLocalizations.incorrect_credentials;
+    } else {
+      _message = appLocalizations.error;
+    }
+    setState(() => _runningQuery = false);
+  }
+
+  // Checks and returns right String if opening mail app is possible
+  String _getSubmitButtonText() {
+    final AppLocalizations appLocalizations = AppLocalizations.of(context)!;
+
+    if (!_send) {
+      return appLocalizations.send_reset_password_mail;
+    } else if (Platform.isAndroid || Platform.isIOS) {
+      return appLocalizations.open_mail_app;
+    } else {
+      return appLocalizations.okay;
+    }
+  }
+
+  // Opens mail app if possible otherwise pops page
+  Future<void> _openMailOrPop() async {
```

Comment:
Yes I also had a look at this url_launcher/mailto combo but for me it looked like it only opens a new email draft which is unnecessary so I stuck with this other package,


> - I feel that a bit intrusive
> - You're not sure that the message will be received at that time

Thats some good arguments, especially the second one since my expierence with off reset emails is that they take very long to arrive. I was just thinking about what to do with the button when the email got send and rememberd apps which automatically read validation sms which I find very comfortable and have therefore checked if there is an easy way to open the email app.

It was no big deal to implement this so I have no problem removing it from the PR if it is not really useful and maybe too compelling

Reply:
Yes, let's just forget about opening the mail app: just have the forgotten password message sent (then dialog "message sent, don't forget to check your email inbox and the spam folder" with "OK" button and that's it), and the end-user will do the rest when the message is received.